packages/node/src/models/free-balance.ts

```diff
@@ -52,14 +55,122 @@ export type TokenIndexedCoinTransferMap = {
   [tokenAddress: string]: CoinTransferMap;
 };
 
+// todo(xuanji): replace with Set
 export type ActiveAppsMap = { [appInstanceIdentityHash: string]: true };
 
-export type FreeBalanceState = {
+export class FreeBalanceClass {
+  private constructor(
+    private readonly activeAppsMap: ActiveAppsMap,
+    private readonly balancesIndexedByToken: {
+      // todo: change this type to TokenIndexedCoinTransferMap
+      [tokenAddress: string]: CoinTransfer[];
+    }
+  ) {}
+  public toFreeBalanceState(): FreeBalanceState {
+    return {
+      activeAppsMap: this.activeAppsMap,
+      balancesIndexedByToken: this.balancesIndexedByToken
+    };
+  }
+  public toTokenIndexedCoinTransferMap() {
+    const ret = {};
+    for (const tokenAddress of Object.keys(this.balancesIndexedByToken)) {
+      ret[tokenAddress] = convertCoinTransfersToCoinTransfersMap(
+        this.balancesIndexedByToken[tokenAddress]
+      );
+    }
+    return ret;
+  }
+  public toAppInstance(oldAppInstance: AppInstance) {
+    return oldAppInstance.setState(
+      serializeFreeBalanceState(this.toFreeBalanceState())
+    );
+  }
+
+  public static createWithFundedTokenAmounts(
+    addresses: string[],
+    amount: BigNumber,
+    tokenAddresses: string[]
+  ): FreeBalanceClass {
+    return new FreeBalanceClass(
+      {},
+      tokenAddresses.reduce(
+        (balancesIndexedByToken, tokenAddress) => ({
+          ...balancesIndexedByToken,
+          [tokenAddress]: addresses.map(to => ({ to, amount }))
+        }),
+        {} as { [tokenAddress: string]: CoinTransfer[] }
+      )
+    );
+  }
+
+  public static fromAppInstance(appInstance: AppInstance): FreeBalanceClass {
+    const freeBalanceState = deserializeFreeBalanceState(
+      appInstance.state as FreeBalanceStateJSON
+    );
+    return new FreeBalanceClass(
+      freeBalanceState.activeAppsMap,
+      freeBalanceState.balancesIndexedByToken
+    );
+  }
+  public getBalance(tokenAddress: string, beneficiary: string) {
+    try {
+      return convertCoinTransfersToCoinTransfersMap(
+        this.balancesIndexedByToken[tokenAddress]
+      )[beneficiary];
+    } catch {
+      return Zero;
```

Comment:
Maybe not entirely necessary but to be explicit it could be useful to clarify in the API doc that calling `getFreeBalance` for a specific token address will (intuitively) return `Zero` even if the channel hasn't ever had any deposits/withdrawals of that token.